rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py

```diff
@@ -0,0 +1,620 @@
+            self.state = RobotState.WAITING
+            with self._lock:
+                if (self.target_waypoint.graph_index is not None):
+                    self.on_waypoint = self.target_waypoint.graph_index
+                else:
+                    self.on_waypoint = None  # we are still on a lane
+                self.last_known_waypoint_index = self.on_waypoint
+
+        def _follow_path():
+            target_pose = []
+            while (
+                    self.remaining_waypoints or
+                    self.state == RobotState.MOVING or
+                    self.state == RobotState.WAITING):
+                # Check if we need to abort
+                if self._quit_path_event.is_set():
+                    self.node.get_logger().info("Aborting previously followed "
+                                                "path")
+                    return
+                # State machine
+                if self.state == RobotState.IDLE:
+                    # Assign the next waypoint
+                    self.target_waypoint = self.remaining_waypoints[0]
+                    self.path_index = self.remaining_waypoints[0].index
+                    # Move robot to next waypoint
+                    target_pose = self.target_waypoint.position
+                    [x, y] = self.transforms["rmf_to_robot"].transform(
+                        target_pose[:2])
+                    theta = target_pose[2] + \
+                        self.transforms['orientation_offset']
+                    # ------------------------ #
+                    # IMPLEMENT YOUR CODE HERE #
+                    # Ensure x, y, theta are in units that api.navigate() #
+                    # ------------------------ #
+                    response = self.api.navigate([x, y, theta], self.map_name)
+
+                    if response:
+                        self.remaining_waypoints = self.remaining_waypoints[1:]
+                        self.state = RobotState.MOVING
+                    else:
+                        self.node.get_logger().info(
+                            f"Robot {self.name} failed to navigate to "
+                            f"[{x:.0f}, {y:.0f}, {theta:.0f}] coordinates. "
+                            f"Retrying...")
+                        time.sleep(0.5)
+
+                elif self.state == RobotState.WAITING:
+                    time.sleep(0.5)
+                    time_now = self.adapter.now()
+                    with self._lock:
+                        if self.target_waypoint is not None:
+                            waypoint_wait_time = self.target_waypoint.time
+                            if (waypoint_wait_time < time_now):
+                                self.state = RobotState.IDLE
+                            else:
+                                if self.path_index is not None:
+                                    d = (waypoint_wait_time - time_now).seconds
+                                    self.node.get_logger().info(
```

Comment:
Lets convert this to `self.node.get_logger().debug(...)`

Reply:
Got it